### PR TITLE
[DOC] ann indexes documentation

### PR DESCRIPTION
**Pull request by gsilvestrin**

### CODE REVIEW

File: docs/src/ann_indexes.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+   for i, row in enumerate(np.random.random((10_000, 768)).astype('float32'))]
+
+# Add the vectors to a table
+tbl = db.create_table("my_vectors", data=data)
+
+# Create and train the index - you need to have enough data in the table for an effective training step
+tbl.create_index(num_partitions=256,num_sub_vectors=96)
+```
+
+Since `create_index` has a training, it can take a couple of minutes to finish for large tables. You can control index
```
Comment (eddyxu) [inline docs/src/ann_indexes.md:28]: => `It usually takes a few minutes to build indices for large tables`?  

"has a training" does not sound a common phrase.

File: docs/src/ann_indexes.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+# Create 10,000 sample vectors
+data = [{"vector": row, "item": f"item {i}"}
+   for i, row in enumerate(np.random.random((10_000, 768)).astype('float32'))]
+
+# Add the vectors to a table
+tbl = db.create_table("my_vectors", data=data)
+
+# Create and train the index - you need to have enough data in the table for an effective training step
+tbl.create_index(num_partitions=256,num_sub_vectors=96)
```
Comment (eddyxu) [inline docs/src/ann_indexes.md:25]: a space after comma.

File: docs/src/ann_indexes.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
+# ANN (Approximate Nearest Neighbor) Indexes
+
+You can create an index over your vector data to make search faster. Vector indexes are faster but less 
+ accurate than exhaustive search. LanceDB provides many parameters to fine-tune the index's size, the speed of 
+queries, and the accuracy of results.
```
Comment (changhiskhan) [inline docs/src/ann_indexes.md:5]: Let's add something like "Currently, LanceDB does not automatically create the ANN index. In the future we will look to improve this experience and automate index creation and configuration"

File: docs/src/ann_indexes.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+   for i, row in enumerate(np.random.random((10_000, 768)).astype('float32'))]
+
+# Add the vectors to a table
+tbl = db.create_table("my_vectors", data=data)
+
+# Create and train the index - you need to have enough data in the table for an effective training step
+tbl.create_index(num_partitions=256, num_sub_vectors=96)
+```
+
+Since `create_index` has a training step, it can take a few minutes to finish for large tables. You can control the index
```
Comment (changhiskhan) [inline docs/src/ann_indexes.md:28]: Provide a good rule of thumb here with examples: number of partitions should be configured so each partition has 3-5K vectors. For example, a table with ~1M vectors should use 256 partitions. You can specify arbitrary number of partitions but powers of 2 is most conventional"

File: docs/src/ann_indexes.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+- **num_partitions**: The number of partitions of the index. A higher number leads to faster queries, but it makes index 
+generation slower.
+- **num_sub_vectors**: The number of subvectors (M) that will be created during Product Quantization (PQ). A larger number makes
+search more accurate, but also makes the index larger and slower to build. 
+
+## Querying an ANN Index
+
+Querying vector indexes is done via the [search](https://lancedb.github.io/lancedb/python/#lancedb.table.LanceTable.search) function.
+
+There are a couple of parameters that can be used to fine-tune the search:
```
Comment (changhiskhan) [inline docs/src/ann_indexes.md:40]: Make sure you mention the defaults

File: docs/src/ann_indexes.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    .nprobes(20) \
+    .refine_factor(20) \
+    .to_df()
+
+                                              vector       item       score
+0  [0.44949695, 0.8444449, 0.06281311, 0.23338133...  item 1141  103.575333
+1  [0.48587373, 0.269207, 0.15095535, 0.65531915,...  item 3953  108.393867
+```
+
+The search will return the data requested in addition to the score of each item. The score is the distance between the query vector and the element. A lower number means that the result is more relevant.
```
Comment (changhiskhan) [inline docs/src/ann_indexes.md:59]: This is missing a section on filtering (the where clause)

**Review by changhiskhan — APPROVED** [review]
